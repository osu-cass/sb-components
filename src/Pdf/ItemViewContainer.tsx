import * as React from 'react'
import { ItemPdfModel, ItemGroupModel } from "./PdfModels";
import { PassageView } from './PassageView';
import { QuestionView } from './QuestionView';

export interface ItemViewContainerProps {
    itemData: ItemGroupModel;
}

export class ItemViewContainer extends React.Component<ItemViewContainerProps, {}> {
    render() {
        const passage = this.props.itemData.passage
            ? <PassageView 
                view={this.props.itemData.passage} 
                associatedItems={this.props.itemData.questions.map(q => q.id)} />
            : null;

        const questions = this.props.itemData.questions.map(q => 
            <QuestionView question={q} key={q.questionNumber} />
        );
        return (
            <div className='page'>
                {passage}
                {questions}
            </div>
        );
    }
}