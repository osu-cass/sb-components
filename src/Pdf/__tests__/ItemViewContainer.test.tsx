import * as React from 'react';
import * as ReactDOM from 'react-dom';
import { ItemViewContainer } from '../ItemViewContainer';
import { shallow, mount, render } from 'enzyme';
import { itemGroup } from './PdfMocks';


describe("EvidenceStatement", () => {
    it("matches snapshot", () => {
        expect(shallow(<ItemViewContainer itemData={itemGroup} />)).toMatchSnapshot();
    })
})
