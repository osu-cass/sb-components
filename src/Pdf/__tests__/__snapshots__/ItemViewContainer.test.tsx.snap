// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EvidenceStatement matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ItemViewContainer
    itemData={
      Object {
        "passage": null,
        "questions": Array [
          Object {
            "data": Object {
              "associatedItems": "string",
              "commonCoreStandardsDescription": "string",
              "depthOfKnowledge": "string",
              "educationalDifficulty": "string",
              "evidenceStatement": "string",
              "itemCardViewModel": Object {
                "bankKey": 1,
                "claimCode": "234536",
                "claimLabel": "Math Claim",
                "commonCoreStandardId": "",
                "depthOfKnowledge": "",
                "domain": "MATH",
                "grade": 8,
                "gradeLabel": "grade 3",
                "interactionTypeCode": "",
                "interactionTypeLabel": "",
                "isPerformanceItem": false,
                "itemKey": 1,
                "subjectCode": "MATH",
                "subjectLabel": "math",
                "targetHash": 212,
                "targetId": "23463467",
                "targetShortName": "1-3",
                "title": "",
              },
              "rubrics": Array [
                Object {
                  "language": "english",
                  "rubricEntries": Array [
                    Object {
                      "name": "",
                      "scorepoint": "",
                      "value": "",
                    },
                  ],
                  "samples": Array [
                    Object {
                      "maxValue": "",
                      "minValue": "",
                      "sampleResponses": Array [
                        Object {
                          "name": "",
                          "purpose": "",
                          "sampleContent": "",
                          "scorePoint": "",
                        },
                      ],
                    },
                  ],
                },
              ],
              "targetDescription": "string",
            },
            "id": "12334",
            "questionNumber": 1,
            "view": Object {
              "captured": true,
              "html": null,
              "id": "12334",
              "picturePath": null,
              "type": 1,
            },
          },
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        null,
        Array [
          <QuestionView
            question={
              Object {
                "data": Object {
                  "associatedItems": "string",
                  "commonCoreStandardsDescription": "string",
                  "depthOfKnowledge": "string",
                  "educationalDifficulty": "string",
                  "evidenceStatement": "string",
                  "itemCardViewModel": Object {
                    "bankKey": 1,
                    "claimCode": "234536",
                    "claimLabel": "Math Claim",
                    "commonCoreStandardId": "",
                    "depthOfKnowledge": "",
                    "domain": "MATH",
                    "grade": 8,
                    "gradeLabel": "grade 3",
                    "interactionTypeCode": "",
                    "interactionTypeLabel": "",
                    "isPerformanceItem": false,
                    "itemKey": 1,
                    "subjectCode": "MATH",
                    "subjectLabel": "math",
                    "targetHash": 212,
                    "targetId": "23463467",
                    "targetShortName": "1-3",
                    "title": "",
                  },
                  "rubrics": Array [
                    Object {
                      "language": "english",
                      "rubricEntries": Array [
                        Object {
                          "name": "",
                          "scorepoint": "",
                          "value": "",
                        },
                      ],
                      "samples": Array [
                        Object {
                          "maxValue": "",
                          "minValue": "",
                          "sampleResponses": Array [
                            Object {
                              "name": "",
                              "purpose": "",
                              "sampleContent": "",
                              "scorePoint": "",
                            },
                          ],
                        },
                      ],
                    },
                  ],
                  "targetDescription": "string",
                },
                "id": "12334",
                "questionNumber": 1,
                "view": Object {
                  "captured": true,
                  "html": null,
                  "id": "12334",
                  "picturePath": null,
                  "type": 1,
                },
              }
            }
          />,
        ],
      ],
      "className": "page",
    },
    "ref": null,
    "rendered": Array [
      null,
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "class",
        "props": Object {
          "question": Object {
            "data": Object {
              "associatedItems": "string",
              "commonCoreStandardsDescription": "string",
              "depthOfKnowledge": "string",
              "educationalDifficulty": "string",
              "evidenceStatement": "string",
              "itemCardViewModel": Object {
                "bankKey": 1,
                "claimCode": "234536",
                "claimLabel": "Math Claim",
                "commonCoreStandardId": "",
                "depthOfKnowledge": "",
                "domain": "MATH",
                "grade": 8,
                "gradeLabel": "grade 3",
                "interactionTypeCode": "",
                "interactionTypeLabel": "",
                "isPerformanceItem": false,
                "itemKey": 1,
                "subjectCode": "MATH",
                "subjectLabel": "math",
                "targetHash": 212,
                "targetId": "23463467",
                "targetShortName": "1-3",
                "title": "",
              },
              "rubrics": Array [
                Object {
                  "language": "english",
                  "rubricEntries": Array [
                    Object {
                      "name": "",
                      "scorepoint": "",
                      "value": "",
                    },
                  ],
                  "samples": Array [
                    Object {
                      "maxValue": "",
                      "minValue": "",
                      "sampleResponses": Array [
                        Object {
                          "name": "",
                          "purpose": "",
                          "sampleContent": "",
                          "scorePoint": "",
                        },
                      ],
                    },
                  ],
                },
              ],
              "targetDescription": "string",
            },
            "id": "12334",
            "questionNumber": 1,
            "view": Object {
              "captured": true,
              "html": null,
              "id": "12334",
              "picturePath": null,
              "type": 1,
            },
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          null,
          Array [
            <QuestionView
              question={
                Object {
                  "data": Object {
                    "associatedItems": "string",
                    "commonCoreStandardsDescription": "string",
                    "depthOfKnowledge": "string",
                    "educationalDifficulty": "string",
                    "evidenceStatement": "string",
                    "itemCardViewModel": Object {
                      "bankKey": 1,
                      "claimCode": "234536",
                      "claimLabel": "Math Claim",
                      "commonCoreStandardId": "",
                      "depthOfKnowledge": "",
                      "domain": "MATH",
                      "grade": 8,
                      "gradeLabel": "grade 3",
                      "interactionTypeCode": "",
                      "interactionTypeLabel": "",
                      "isPerformanceItem": false,
                      "itemKey": 1,
                      "subjectCode": "MATH",
                      "subjectLabel": "math",
                      "targetHash": 212,
                      "targetId": "23463467",
                      "targetShortName": "1-3",
                      "title": "",
                    },
                    "rubrics": Array [
                      Object {
                        "language": "english",
                        "rubricEntries": Array [
                          Object {
                            "name": "",
                            "scorepoint": "",
                            "value": "",
                          },
                        ],
                        "samples": Array [
                          Object {
                            "maxValue": "",
                            "minValue": "",
                            "sampleResponses": Array [
                              Object {
                                "name": "",
                                "purpose": "",
                                "sampleContent": "",
                                "scorePoint": "",
                              },
                            ],
                          },
                        ],
                      },
                    ],
                    "targetDescription": "string",
                  },
                  "id": "12334",
                  "questionNumber": 1,
                  "view": Object {
                    "captured": true,
                    "html": null,
                    "id": "12334",
                    "picturePath": null,
                    "type": 1,
                  },
                }
              }
            />,
          ],
        ],
        "className": "page",
      },
      "ref": null,
      "rendered": Array [
        null,
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "class",
          "props": Object {
            "question": Object {
              "data": Object {
                "associatedItems": "string",
                "commonCoreStandardsDescription": "string",
                "depthOfKnowledge": "string",
                "educationalDifficulty": "string",
                "evidenceStatement": "string",
                "itemCardViewModel": Object {
                  "bankKey": 1,
                  "claimCode": "234536",
                  "claimLabel": "Math Claim",
                  "commonCoreStandardId": "",
                  "depthOfKnowledge": "",
                  "domain": "MATH",
                  "grade": 8,
                  "gradeLabel": "grade 3",
                  "interactionTypeCode": "",
                  "interactionTypeLabel": "",
                  "isPerformanceItem": false,
                  "itemKey": 1,
                  "subjectCode": "MATH",
                  "subjectLabel": "math",
                  "targetHash": 212,
                  "targetId": "23463467",
                  "targetShortName": "1-3",
                  "title": "",
                },
                "rubrics": Array [
                  Object {
                    "language": "english",
                    "rubricEntries": Array [
                      Object {
                        "name": "",
                        "scorepoint": "",
                        "value": "",
                      },
                    ],
                    "samples": Array [
                      Object {
                        "maxValue": "",
                        "minValue": "",
                        "sampleResponses": Array [
                          Object {
                            "name": "",
                            "purpose": "",
                            "sampleContent": "",
                            "scorePoint": "",
                          },
                        ],
                      },
                    ],
                  },
                ],
                "targetDescription": "string",
              },
              "id": "12334",
              "questionNumber": 1,
              "view": Object {
                "captured": true,
                "html": null,
                "id": "12334",
                "picturePath": null,
                "type": 1,
              },
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
